docs/design/specs/Ecma-335-Augments.md

```diff
@@ -430,7 +433,8 @@ The algorithm is amended as follows:
   * The original step 4 is moved after the following steps:
     * Create an empty list of candidate implementations of the interface method.
     * If the interface method itself is not abstract, add it to the list.
-    * Apply all MethodImpls specified in the list of interfaces implicitly implemented by the runtime class of the instance through which the interface method is invoked and add the methods to the list.
+    * For instance methods, apply all MethodImpls specified in the list of interfaces implicitly implemented by the runtime class of the instance through which the interface method is invoked and add the methods to the list.
+    * For static virtual methods, apply all MethodImpls specified in the list of interfaces implicitly implemented by the type used in the `constrained.` prefix of the `call` / `ldftn` instruction and add the methods to the list.
```

Comment:
> implicitly implemented by the type  [](http://example.com/codeflow?start=91&length=36)

I am not sure how to interpret the "implicitly implemented interface" term. Are there more than one way for a type to implement an interface? I see that this terminology was previously used for instance methods. So, this is more like a curiosity question.

Reply:
Yes. An interface can be implemented explicitly (where the interface exists in the interfaceimpl table for the type directly) or implicitly (where the interface is implemented, but only because the base type implemented the interface). This doesn't really matter for the semantics when Roslyn generates code, as Roslyn always redeclares all of the interfaces, but at runtime we have to handle the case where the base type has been changed since the derived type was had its IL generated by Roslyn. For... reasons, the behavior is slightly different.